src/cargo/ops/cargo_compile.rs

```diff
@@ -661,12 +661,24 @@ pub fn create_bcx<'a, 'cfg>(
                 continue;
             }
 
+            let guidance = if ws.is_ephemeral() {
```

Comment:
Could we make the assumption that this is `cargo install` and tell users to run with `--locked` as the lock file is likely to point to rust-version compatible dependencies?

@ehuss mentioned an `ignore_lock` and the name sounds like it'd be for `cargo install` without `--locked` so we'd probably want to include that in the condition for this message.

Reply:
@epage I wonder if, in a new edition, we could switch the default of `cargo install` to `--locked`.

Meanwhile, sure, I can do that.